lib/moonshot/stack_parameter.rb

```diff
@@ -0,0 +1,64 @@
+module Moonshot
+  class StackParameter
+    attr_reader :name
+    attr_reader :default
+
+    def initialize(name, default: nil, use_previous: false)
+      @name = name
+      @default = default
+      @use_previous = use_previous
+      @value = nil
+    end
+
+    # Does this Stack Parameter have a default value that will be used?
+    def default?
+      !@default.nil?
+    end
+
+    def use_previous?
+      @use_previous ? true : false
```

Comment:
I don't see the scenario where this can be `nil`. Even if that could occur, you can still use `!!@use_previous`. So I wonder why the ternary 🤔